iceoryx_posh/source/roudi/roudi.cpp

```diff
@@ -234,6 +234,8 @@ void RouDi::processMessage(const runtime::IpcMessage& message,
             capro::ServiceDescription service(cxx::Serialization(message.getElementAtIndex(2)));
             cxx::Serialization portConfigInfoSerialization(message.getElementAtIndex(7));
 
+            cxx::Expects(service.isValid() && "invalid service description!");
```

Comment:
Okay, this is exactly the place where we shouldn't use `cxx::Expects`. It's an input from a user application we potentially cannot trust and we need to assume that it is indeed invalid. Please replace it with a check similar to three lines below

Reply:
:see_no_evil: Of course you're right. Fixed.